src/lib/allele_set.ml

```diff
@@ -0,0 +1,48 @@
+
+open Util
+
+module SMap = Map.Make (struct type t = string let compare = String.compare end)
+module IMap = Map.Make (struct type t = int let compare = Int.compare end)
+
+type ds =
+  { size      : int
+  ; to_index  : int SMap.t
+  ; to_allele : string IMap.t
+  }
+
+let construct lst =
+  List.sort ~cmp:compare lst
+  |> List.fold_left ~init:(0, SMap.empty, IMap.empty)
+      ~f:(fun (i, sm, im) a -> (i + 1, SMap.add a i sm, IMap.add i a im))
+  |> fun (size, to_index, to_allele) ->
```

Comment:
Is this reason you're packing this into a record here and not in the accumulator because it's faster to construct a tuple than a record?